lib/pipeline/RequestHandler.js

```diff
@@ -1,6 +1,8 @@
 'use strict';
 
 const Duplex  = require('readable-stream').Duplex;
+const agent = require('../agent');
+const logger = agent.logger;
 
 class RequestHandler extends Duplex {
   constructor(options) {
```

Comment:
Why not pass the logger as a dependency?

Reply:
Because it has been always a singleton and this class is not exported. Didn't make too much sense to keep it like that